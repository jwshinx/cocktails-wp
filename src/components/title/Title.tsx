import React from 'react'

import styles from './Title.module.css'

export default function Title({ text }: { text: string }) {
  return (
    <div className={`${styles.title}`}>
      <h2 data-testid="title">{text}</h2>
    </div>
  )
}
