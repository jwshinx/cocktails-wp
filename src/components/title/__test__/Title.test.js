/**
 * @jest-environment jsdom
 */

import React from 'react'
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import Title from '../Title'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Title></Title>, div)
})

it('renders title properly', () => {
  const { getByTestId } = render(<Title text="Foobar"></Title>)
  expect(getByTestId('title')).toHaveTextContent('Foobar')
})

it('renders title properly', () => {
  const { getByTestId } = render(<Title text="Cocktails"></Title>)
  expect(getByTestId('title')).toHaveTextContent('Cocktails')
})

it('should snapshot render Cocktails', () => {
  const tree = renderer.create(<Title text="Cocktails"></Title>).toJSON()
  expect(tree).toMatchSnapshot()
})
