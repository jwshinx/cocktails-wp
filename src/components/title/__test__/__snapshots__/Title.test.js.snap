// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="title"
>
  <h2
    data-testid="title"
  >
    Cocktails
  </h2>
</div>
`;
